addons/barcodes/static/src/barcode_service.js

```diff
@@ -0,0 +1,153 @@
+            }
+            if (barcodeInput) {
+                blurBarcodeInput();
+            }
+            currentStr = "";
+            currentTarget = null;
+        }
+
+        function keydownHandler(ev) {
+            // Don't catch non-printable keys for which Firefox triggers a keypress
+            if (isSpecialKey(ev.key)) {
+                return;
+            }
+            // Don't catch keypresses which could have a UX purpose (like shortcuts)
+            if (ev.ctrlKey || ev.metaKey || ev.altKey) {
+                return;
+            }
+
+            currentTarget = ev.target;
+            // Don't catch events targeting elements that are editable because we
+            // have no way of redispatching 'genuine' key events. Resent events
+            // don't trigger native event handlers of elements. So this means that
+            // our fake events will not appear in eg. an <input> element.
+            if (currentTarget !== barcodeInput && isEditable(currentTarget) &&
+                !currentTarget.dataset.enableBarcode &&
+                currentTarget.getAttribute("barcode_events") !== "true") {
+                return;
+            }
+
+            if (ev.key !== "Enter" && ev.key !== "Unidentified") {
+                currentStr += ev.key;
+            }
+            clearTimeout(timeout);
+            if (String.fromCharCode(ev.which).match(endRegexp)) {
+                checkBarcode();
+            } else {
+                timeout = setTimeout(checkBarcode, barcodeService.maxTimeBetweenKeysInMs);
+            }
+        }
+
+        function mobileChromeHandler(ev) {
+            if (ev.key === "Unidentified") {
+                return;
+            }
+            if ($(document.activeElement).not('input:text, textarea, [contenteditable], ' +
+                '[type="email"], [type="number"], [type="password"], [type="tel"], [type="search"]').length) {
+                barcodeInput.focus();
+            }
+            keydownHandler(ev);
+            clearTimeout(blurTimeout);
+            // if the barcode input doesn't receive keydown for a while, remove it.
+            blurTimeout = setTimeout(() => blurBarcodeInput(), inputTimeOut)
+        }
+
+        function blurBarcodeInput() {
+            barcodeInput.value = "";
+            // fixme: blurring the input should not be necessary, but for some
+            // reason, doing it fix some weird scrolling issue in a mobile (non
+            // native) device. 
+            barcodeInput.blur();
+        }
+
+        whenReady(() => {
+            const isMobileChrome = barcodeService.isMobileChrome;
+            if (isMobileChrome) {
+                barcodeInput = makeBarcodeInput();
+                document.body.appendChild(barcodeInput);
+            }
+            const handler = isMobileChrome ? mobileChromeHandler : keydownHandler;
+            document.body.addEventListener('keydown', handler);
```

Comment:
Seems that mobile handler now uses 'keydownHandler' and it  wasn't the case before. So, I'm wondering if we can squash all of this and use the same handler...